aidial_adapter_bedrock/embedding/amazon/titan_image.py

```diff
@@ -0,0 +1,163 @@
+            "inputImage": request.inputImage,
+            "dimensions": dimensions,
+        }
+    )
+
+
+async def download_image(
+    attachment: Attachment, storage: FileStorage | None
+) -> str:
+    _content_type, data = await download_base64_data(
+        attachment, storage, ["image/png"]
+    )
+    return data
+
+
+def get_requests(
+    request: EmbeddingsRequest, storage: FileStorage | None
+) -> AsyncIterator[AmazonRequest]:
+    async def on_text(text: str):
+        return AmazonRequest(inputText=text)
+
+    async def on_attachment(attachment: Attachment):
+        return AmazonRequest(
+            inputImage=await download_image(attachment, storage)
+        )
+
+    async def on_text_or_attachment(text: str | Attachment):
+        if isinstance(text, str):
+            return await on_text(text)
+        else:
+            return await on_attachment(text)
+
+    async def on_mixed(inputs: List[str | Attachment]) -> AmazonRequest:
+        if len(inputs) == 0:
+            raise EMPTY_INPUT_LIST_ERROR
+        elif len(inputs) == 1:
+            return await on_text_or_attachment(inputs[0])
+        elif len(inputs) == 2:
+            if isinstance(inputs[0], str) and isinstance(inputs[1], Attachment):
+                return AmazonRequest(
+                    inputText=inputs[0],
+                    inputImage=await download_image(inputs[1], storage),
+                )
+            elif isinstance(inputs[0], Attachment) and isinstance(
+                inputs[1], str
+            ):
+                return AmazonRequest(
+                    inputText=inputs[1],
+                    inputImage=await download_image(inputs[0], storage),
+                )
+            else:
+                raise ValidationError(
+                    "The first element of a custom_input list element must be a string and the second element must be an image attachment or vice versa"
+                )
+        else:
+            raise ValidationError(
+                "No more than two elements are allowed in an element of custom_input list"
+            )
+
+    return collect_embedding_inputs(
+        request,
+        on_text=on_text,
+        on_attachment=on_attachment,
+        on_mixed=on_mixed,
+    )
+
+
+class AmazonResponse(BaseModel):
+    inputTextTokenCount: int
+    embedding: List[float]
+
+
+class AmazonTitanImageEmbeddings(EmbeddingsAdapter):
+    model: str
+    client: Bedrock
+    storage: FileStorage | None
+
+    @classmethod
+    def create(cls, client: Bedrock, model: str, api_key: str) -> Self:
+        storage = create_file_storage(api_key)
+        return cls(client=client, model=model, storage=storage)
+
+    async def embeddings(
+        self, request: EmbeddingsRequest
+    ) -> EmbeddingsResponse:
+
+        # The model in fact does not support dimensions,
+        # but the documentation claims it does
+        validate_embeddings_request(request, supports_dimensions=False)
```

Comment:
I've tested, and looks like they does
Just dimensions parameter migrated to `"embeddingConfig": {"outputEmbeddingLength": dimensions},` parameter

This test script works for me
```python
import boto3
import base64
import json

bedrock = boto3.client(service_name="bedrock-runtime")


with open("./sample_image.png", "rb") as file:
    content = file.read()

encoded_image = base64.b64encode(content).decode("utf-8")


def test_embedding_dimensions(dimensions: int):

    body = json.dumps(
        {
            "inputImage": base64.b64encode(content).decode("utf-8"),
            "embeddingConfig": {"outputEmbeddingLength": dimensions},
        }
    )

    response = bedrock.invoke_model(
        body=body,
        modelId="amazon.titan-embed-image-v1",
        accept="application/json",
        contentType="application/json",
    )
    response_body = json.loads(response.get("body").read())

    assert len(response_body["embedding"]) == dimensions
    print("Dimensions match with response!")


test_embedding_dimensions(256)
test_embedding_dimensions(1024)
test_embedding_dimensions(384)
```

Reply:
Indeed, fixed. Good catch!